lib/pengine/bundle.c

```diff
@@ -1585,6 +1663,72 @@ pe__print_bundle(pe_resource_t *rsc, const char *pre_text, long options,
     }
 }
 
+void
+pe__output_bundle(pe_resource_t *rsc, long options, void * out)
+{
+    pcmk__output_t * _out = (pcmk__output_t *)out;
+    pe__bundle_variant_data_t *bundle_data = NULL;
+    char *child_text = NULL;
+    char buffer[LINE_MAX];
+    CRM_CHECK(rsc != NULL, return);
+
+    if (options & pe_print_xml) {
+        bundle_output_xml(rsc, options, out);
+        return;
+    }
+
+    get_bundle_variant_data(bundle_data, rsc);
+
+    snprintf(buffer, LINE_MAX, "Container bundle%s: %s [%s]%s%s\n",
+                 ((bundle_data->nreplicas > 1)? " set" : ""),
+                 rsc->id, bundle_data->image,
+                 is_set(rsc->flags, pe_rsc_unique) ? " (unique)" : "",
+                 is_set(rsc->flags, pe_rsc_managed) ? "" : " (unmanaged)");
+    _out->list_item(_out, NULL, buffer);
+    if (options & pe_print_html) {
+        _out->list_item(_out, "br", NULL);
+        _out->end_list(_out); //ul
+    }
```

Comment:
For example, here, we would not use the generic list_item() method, but instead define a custom output function and call it here. The text implementation would print the buffer as above, the HTML implementation (not needed in this PR) would do the same and add whatever tags are desired, and the XML implementation would add XML attributes.

As an example, see how lib/fencing/st_output.c defines fence_target_text() and fence_target_xml() then registers them as the implementation of the "fence-target" custom output message.

Here, it might make sense to define a pe_output.c that declares whatever custom messages we need, along with a pe__register_messages() function to register them. (The actual functions would be as you have them here.)

Reply:
I will add a "new-line" message that should output a `<br>` tag when printing the result.

Reply:
For html